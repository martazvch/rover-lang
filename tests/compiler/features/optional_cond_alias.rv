fn getValue() -> ?float { return 5. }

fn main() {
    if getValue() as v {
        print v
    }

    while getValue() as v {
        print v
    }
}
