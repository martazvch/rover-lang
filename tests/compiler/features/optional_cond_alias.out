-- getValue --
constant index 0, value 5
ret

-- main --
load_sym index 0, <function getValue>
call index 0
ne_null_push
jump_false 5 -> 16
pop
get_local index 0
print
pop
jump 13 -> 18

// Pops if condition and lhs of extractor
pop
pop

// While
load_sym index 0, <function getValue>
call index 0
ne_null_push
jump_false 23 -> 34
pop
get_local index 0
print
pop
loop 31 -> 18

// Pops if condition and lhs of extractor
pop
pop

ret_naked

-- global scope --
load_sym index 1, <function main>
call index 0
